source/libnormaliz/dynamic_bitset.h

```diff
@@ -24,12 +24,449 @@
+        for (limb_t& limb: _limbs) {
+            limb ^= all_ones_limb;
+        }
+        cleanup_last_limb(); // cleanup in the last limb
+        return *this;
+    }
+
+    // return true if any bit is set to 1, otherwise false
+    bool any() const {
+        for (const limb_t& limb: _limbs) {
+            if (limb != 0)
+                return true;
+        }
+        return false;
+    }
+
+    // return true if no bit is set to 1, otherwise false
+    bool none() const {
+        return !any();
+    }
+
+    // return the number of bits which are set to 1
+    size_t count() const {
+        size_t sum = 0;
+        for (const limb_t& limb: _limbs) {
+            sum += popcount(limb);
+        }
+        return sum;
+    }
+
+
+    //
+    // in-place operators
+    //
+    dynamic_bitset& operator&=(const dynamic_bitset& rhs) {
+        assert(size() == rhs.size());
+        for (size_t i = 0; i < _limbs.size(); ++i) {
+            _limbs[i] &= rhs._limbs[i];
+        }
+        return *this;
+    }
+    dynamic_bitset& operator|=(const dynamic_bitset& rhs) {
+        assert(size() == rhs.size());
+        for (size_t i = 0; i < _limbs.size(); ++i) {
+            _limbs[i] |= rhs._limbs[i];
+        }
+        return *this;
+    }
+    dynamic_bitset& operator^=(const dynamic_bitset& rhs) {
+        assert(size() == rhs.size());
+        for (size_t i = 0; i < _limbs.size(); ++i) {
+            _limbs[i] ^= rhs._limbs[i];
+        }
+        return *this;
+    }
+    dynamic_bitset& operator-=(const dynamic_bitset& rhs) {
+        assert(size() == rhs.size());
+        for (size_t i = 0; i < _limbs.size(); ++i) {
+            _limbs[i] &= ~rhs._limbs[i];
+        }
+        return *this;
+    }
+
+
+    //
+    // non-modifying unary and binary operators
+    //
+
+    dynamic_bitset operator~() const  {
+        dynamic_bitset result(*this);
+        result.flip();
+        return result;
+    }
+    dynamic_bitset operator&(const dynamic_bitset& rhs) const {
+        dynamic_bitset result(*this);
+        return result &= rhs;
+    }
+    dynamic_bitset operator|(const dynamic_bitset& rhs) const {
+        dynamic_bitset result(*this);
+        return result |= rhs;
+    }
+    dynamic_bitset operator^(const dynamic_bitset& rhs) const {
+        dynamic_bitset result(*this);
+        return result ^= rhs;
+    }
+    dynamic_bitset operator-(const dynamic_bitset& rhs) const {
+        dynamic_bitset result(*this);
+        return result -= rhs;
+    }
+
+    //
+    // comparison of two bitsets
+    //
+
+    bool operator==(const dynamic_bitset &x) const {
+        return _total_bits == x._total_bits && _limbs == x._limbs;
+    }
+    bool operator!=(const dynamic_bitset &x) const {
+        return !(*this == x);
+    }
+    bool operator<(const dynamic_bitset &x) const {    // for use in std::set, std::map
+        // FIXME: the following is probably not compatible with boost::dyanmic_bitset; do we care?
```

Comment:
This is the only TODO/FIXME comment left... I think we only need this method so that we can put dynamic_bitset instances into `std::set` resp. use them as keys in `std::map`, so the order shouldn't matter -- should it?